File: iOS/ApiTestingUtil/ApiTestingUtil/Interface/Model/ApiProject.swift
//
//  ApiProject.swift
//  ApiTestingUtil
//
//  Created by Jason Lee on 16/01/25.
//  Copyright © 2016年 CZLee. All rights reserved.
//

import UIKit

struct ApiProject {
    private var projectName: String?
    private var baseURL: String?
    private var routeURL: String?
    private var apiArray: [ApiInterface] = Array()
    
    var name: String {
        get {
            return self.projectName ?? ""
        }
    }
    var base_url: String {
        get {
            return self.baseURL ?? ""
        }
    }
    var route_url: String {
        get {
            return self.routeURL ?? ""
        }
    }
    var apis: [ApiInterface] {
        return self.apiArray
    }
    
    init(data: [String: AnyObject]) {
        self.projectName = data["project_name"] as? String
        self.baseURL = data["base_url"] as? String
        self.routeURL = data["route_url"] as? String
        if let tempArray = data["apis"] as? [[String : AnyObject]] {
            for data in tempArray {
                self.apiArray.append(ApiInterface(data: data))
            }
        }
    }
    
    func toDictionary() -> [String : AnyObject] {
        return ["project_name" : self.name,
            "base_url" : self.base_url,
            "route_url" : self.route_url,
            //            "apis" : self.apiArray
        ]
    }
    
    func toString() -> String {
        var apiString = ""
        for api in self.apiArray {
            apiString += "\(api.toString())"
        }
        return ("{\nproject_name : \(self.name)\nbase_url : \(self.base_url)\nroute_url : \(self.route_url)\napis : \n\(apiString)\n}")
    }
}

File: Android/ApiTestUtil/app/src/main/java/cn/com/hvit/apitestutil/Interface/Bean/ApiProject.java
package cn.com.hvit.apitestutil.Interface.Bean;

import android.util.Log;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

/**
 * Created by JasonLee on 16/01/25.
 */
public class ApiProject {
    public String project_name;
    public String base_url;
    public String route_url;
    public ArrayList<ApiInterface> apiArray;

    public void init(Object data) {
        HashMap<String, Object> project = (HashMap<String, Object>) data;
        if (project != null) {
            this.project_name = project.get("project_name").toString();
            this.base_url = project.get("base_url").toString();
            this.route_url = project.get("route_url").toString();
            List<Object> tempArray = (List<Object>) project.get("apis");
            this.apiArray = new ArrayList<ApiInterface>();
            for (Object object :
                    tempArray) {
                ApiInterface api = new ApiInterface();
                api.init(object);
                this.apiArray.add(api);
            }
        }
    }

    public Object toObject() {
        HashMap<String, Object> object = new HashMap<String, Object>();

        object.put("project_name", project_name);
        object.put("base_url", base_url);
        object.put("route_url", route_url);
        object.put("apis", apiArray);

        return object;
    }

    public void logString() {
        Log.i("project_name", this.project_name);
        Log.i("base_url", this.base_url);
        Log.i("route_url", this.route_url);
        Log.i("apis", "");
        for (ApiInterface api :
                this.apiArray) {
            api.logString();
        }
    }
}
